src/einsteinpy/metric/kerrnewman.py

```diff
@@ -1,89 +1,418 @@
+            drh2dth = -2 * (alpha ** 2) * np.cos(th) * np.sin(th)
+            dgdx[2, 0, 0] = (
+                (-2 * (alpha ** 2) * np.sin(th) * np.cos(th)) * rho2
+                - drh2dth * (dl - ((alpha * np.sin(th)) ** 2))
+            ) / (rho2 ** 2)
+            dgdx[2, 1, 1] = -drh2dth / (c2 * dl)
+            dgdx[2, 2, 2] = -drh2dth / c2
+            dgdx[2, 3, 3] = (1 / (c2 * (rho2 ** 2))) * (
+                (
+                    (
+                        (4 * (alpha ** 2) * (np.sin(th) ** 3) * np.cos(th) * dl)
+                        - (2 * np.sin(th) * np.cos(th) * ((r ** 2 + alpha ** 2) ** 2))
+                    )
+                    * rho2
+                )
+                - (
+                    drh2dth
+                    * (((alpha * np.sin(th)) ** 2) * dl - ((r ** 2 + alpha ** 2) ** 2))
+                    * (np.sin(th) ** 2)
+                )
+            )
+            dgdx[2, 0, 3] = dgdx[2, 3, 0] = (
+                (-alpha * (dl - r ** 2 - alpha ** 2)) / (_c * (rho2 ** 2))
+            ) * ((2 * np.sin(th) * np.cos(th) * rho2) - (drh2dth * (np.sin(th) ** 2)))
+
+        due_to_r()
+        due_to_theta()
+
+        return dgdx
+
+    def _dg_dx_ks(self, x_vec):
+        """
+        Returns derivative of each Kerr-Newman Metric component \
+        w.r.t. coordinates in Kerr-Schild Coordinate System
+
+        Parameters
+        ----------
+        x_vec : ~numpy.ndarray
+                Position 4-Vector
+
+        Returns
+        -------
+        NotImplementedError
+            To be implemented after KS Coordinates
+
+        """
+        # To be implemented after KS Coordinates
+        raise NotImplementedError
+
+    def _christoffels(self, x_vec):
+        """
+        Returns Christoffel Symbols for Kerr-Newman Metric in chosen Coordinates
+
+        Parameters
+        ----------
+        x_vec : ~numpy.ndarray
+            Position 4-Vector
+
+        Returns
+        -------
+        ~numpy.ndarray
+            Christoffel Symbols for Kerr-Newman \
+            Metric in chosen Coordinates
+            Numpy array of shape (4,4,4)
+
+        """
+        if self.coords == "KS":
+            return self._ch_sym_ks(x_vec)
+
+        return self._ch_sym_bl(x_vec)
+
+    def _ch_sym_bl(self, x_vec):
+        """
+        Returns Christoffel Symbols for Kerr-Newman Metric \
+        in Boyer-Lindquist Coordinates
+
+        Parameters
+        ----------
+        x_vec : ~numpy.ndarray
+            Position 4-Vector
+
+        Returns
+        -------
+        ~numpy.ndarray
+            Christoffel Symbols for Kerr-Newman Metric \
+            in Boyer-Lindquist Coordinates
+            Numpy array of shape (4,4,4)
+
+        """
+        g_contra = self.metric_contravariant(x_vec)
+        dgdx = self._dg_dx_bl(x_vec)
+
+        chl = np.zeros(shape=(4, 4, 4), dtype=float)
+
+        for _, k, l in self.nonzero_christoffels_list_bl[0:4]:
+            val1 = dgdx[l, 0, k] + dgdx[k, 0, l]
```

Comment:
Why `_`  instead of something more descriptive?

Reply:
This denotes a throwaway variable. As you can see, `_` is not used in the body of the loop. It's only there for proper list comprehension of `self.nonzero_christoffels_list_bl[0:4]` into `k` and `l`.